### DISCUSSION TITLE
DropdownEdit

### DISCUSSION THREAD

**Discussion by medhome63** [mse-org/mseuniverse#5] [Q&A]
Hi All,
 
I am working with tdropdownlistEditLB combined with tdblookupbuffer and  I did not find the object propertie where the index and the array of data items are stored when the event onChange is fired.

There is not a dropdown.cols[n].value or a  function to get ( n, value).  Has anyone worked on this before to help us is this subject?
 
Med.

Ps: I  noticed that onFilter event gives paramerters to get index and value ?
 procedure tmainfo.onFilterEV(const sender: tcustomlookupbuffer, const physindex: Integer; var valid: Boolean);
and I want to do: tmsesqlQuery( datasource.dataset).indexlocal(0). find(   value of column (n) index physindex.)

**Comment by fredvs** [comment]
Hello Med.

About _tdropdownlistEditLB_, indeed it seems that it was not yet totally completed and should be revisited.
I will jump into your sample_lookup.zip asap (I am super busy this week :-( )
Fred

**Comment by medhome63** [comment]
Hello Fred,

The goal is not to change   tdropdownlisteditlb.value but to get  ( nRow, nCol , value) of the selected item.

Thank you, Med


________________________________
De : Fred vS ***@***.***>
Envoyé : mercredi 6 septembre 2023 13:25
À : mse-org/mseuniverse ***@***.***>
Cc : Med Hamza ***@***.***>; Author ***@***.***>
Objet : Re: [mse-org/mseuniverse] DropdownEdit (Discussion #5)


Hello Med.

About tdropdownlistEditLB, indeed it seems that it was not yet totally completed and should be revisited.
I will jump into your sample_lookup.zip asap (I am super busy this week :-( )
Fred

**Comment by fredvs** [comment]
Hello Med.

Yes, I understood this but the fact is that I dont see how to get the column selected.
And maybe the code of tdropdownlisteditlb not permit it like it is now.

Fre;D

________________________________
De : Med Hamza ***@***.***>
Envoyé : vendredi 8 septembre 2023 20:50
À : mse-org/mseuniverse ***@***.***>
Cc : Fred vS ***@***.***>; Comment ***@***.***>
Objet : Re: [mse-org/mseuniverse] DropdownEdit (Discussion #5)


Hello Fred,

The goal is not to change tdropdownlisteditlb.value but to get ( nRow, nCol , value) of the selected item.

Thank you, Med


________________________________
De : Fred vS ***@***.***>
Envoyé : mercredi 6 septembre 2023 13:25
À : mse-org/mseuniverse ***@***.***>
Cc : Med Hamza ***@***.***>; Author ***@***.***>
Objet : Re: [mse-org/mseuniverse] DropdownEdit (Discussion #5)


Hello Med.

About tdropdownlistEditLB, indeed it seems that it was not yet totally completed and should be revisited.
I will jump into your sample_lookup.zip asap (I am super busy this week :-( )
Fred

**Comment by medhome63** [comment]
Hello fred,
  You can notice that we can have any column, but the problem is the row number selected ?

for example:
  var nCol:integer;
  ....
  nCol:=1;

  procedure tmainfo.onchange ()
  var nRow:integer;

begin
    nrow:=??????   <-----------  REWARD
    showmessage(lookuplb.textvalue[nCol, nRow]);
end;

Regards

Med.






________________________________
De : Fred vS ***@***.***>
Envoyé : vendredi 8 septembre 2023 19:34
À : mse-org/mseuniverse ***@***.***>
Cc : Med Hamza ***@***.***>; Author ***@***.***>
Objet : Re: [mse-org/mseuniverse] DropdownEdit (Discussion #5)


Hello Med.

Yes, I understood this but the fact is that I dont see how to get the column selected.
And maybe the code of tdropdownlisteditlb not permit it like it is now.

Fre;D

________________________________
De : Med Hamza ***@***.***>
Envoyé : vendredi 8 septembre 2023 20:50
À : mse-org/mseuniverse ***@***.***>
Cc : Fred vS ***@***.***>; Comment ***@***.***>
Objet : Re: [mse-org/mseuniverse] DropdownEdit (Discussion #5)


Hello Fred,

The goal is not to change tdropdownlisteditlb.value but to get ( nRow, nCol , value) of the selected item.

Thank you, Med


________________________________
De : Fred vS ***@***.***>
Envoyé : mercredi 6 septembre 2023 13:25
À : mse-org/mseuniverse ***@***.***>
Cc : Med Hamza ***@***.***>; Author ***@***.***>
Objet : Re: [mse-org/mseuniverse] DropdownEdit (Discussion #5)


Hello Med.

About tdropdownlistEditLB, indeed it seems that it was not yet totally completed and should be revisited.
I will jump into your sample_lookup.zip asap (I am super busy this week :-( )
Fred

**Comment by fredvs** [comment]
Hello Med.

Sorry but I did not find yet time to jump into it.

I will try to find time tonight.

Write you later.

Fre;D

**Comment by medhome63** [comment]
Hello Fred,

 Ultimately it's  simplier than we thought .

Just:
ncol:=1;  for example

procedure tmainfo.changeEV(const sender: TObject);
var s:string;
begin
with tdropdownlisteditlb1 do
s:=(lookuplb.textvalue[ncol,dropdown.itemindex]);

// and we can do here a dataset find ( s);
end;

Regards.

Med
PS:itemidex is in msedataedits.pas


________________________________
De : Fred vS ***@***.***>
Envoyé : dimanche 10 septembre 2023 16:04
À : mse-org/mseuniverse ***@***.***>
Cc : Med Hamza ***@***.***>; Author ***@***.***>
Objet : Re: [mse-org/mseuniverse] DropdownEdit (Discussion #5)


Hello Med.

Sorry but I did not find yet time to jump into it.

I will try to find time tonight.

Write you later.

Fre;D

**Comment by fredvs** [comment]
Hello Med.

Ha, ok, nice you found the solution.


Huh, may I ask you to update the [sample_lookup.zip](https://github.com/mse-org/mseuniverse/files/12535730/sample_lookup.zip) file and include the missing **_main.mfm_** file?

(So I can better understand what was the question and the solution).

Thanks.

Fre;D

**Comment by fredvs** [comment]
Ok,  but the solution how to get the selected column if number of columns > 1 is still mystery.

But that is a other story and maybe not really needed ( if you decide before popup what column must be retrieved ).

**Comment by medhome63** [comment]
Hello Fred,

Here is the the complete demo prj. Notice that we can use the same datasource for the lookupbuffer and the grid.

See also lookup dboptions  which is important in case of adding record to dataset.

Med

________________________________
De : Fred vS ***@***.***>
Envoyé : dimanche 10 septembre 2023 18:42
À : mse-org/mseuniverse ***@***.***>
Cc : Med Hamza ***@***.***>; Author ***@***.***>
Objet : Re: [mse-org/mseuniverse] DropdownEdit (Discussion #5)


Ok, but the solution how to get the selected column if number of columns > 1 is still mystery.

But that is a other story and maybe not really needed ( if you decide before popup what column must be retrieved ).

**Comment by fredvs** [comment]
Hello Med.

>Here is the the complete demo prj. 

Hum, where, I dont see any attachment... ?!

Fre;D

**Reply by medhome63** [reply]
[sample_lookup.zip](https://github.com/mse-org/mseuniverse/files/12588424/sample_lookup.zip)

**Comment by medhome63** [comment]
Re-Hello

It's  sample_lookup.zip I can see in my box?

May be I have to upload it on GitHub ? not as attn in hotmail?

Med.
________________________________
De : Fred vS ***@***.***>
Envoyé : mardi 12 septembre 2023 15:23
À : mse-org/mseuniverse ***@***.***>
Cc : Med Hamza ***@***.***>; Author ***@***.***>
Objet : Re: [mse-org/mseuniverse] DropdownEdit (Discussion #5)


Hello Med.

Here is the the complete demo prj.

Hum, where, I dont see any attachment... ?!

Fre;D

**Comment by fredvs** [comment]
Yes, please, upload it on Github, I dont get the attachment..

You may also edit your post in Github ( click on the 3 dots ... ), delete your initial zip and upload the new.

**Comment by fredvs** [comment]
Ha, ok, I see it now.
Thanks.

**Reply by medhome63** [reply]
OK. Would you like to upload it in mseunivers\sample\db. I could not create a new directory!!!


MED

**Reply by fredvs** [reply]
OK, I will do.
Many thanks for that new demo.

PS: To create a new directory, you need to do it via git, first clone mseuniverse and do the change in your local machine, then push it to github.

```
git clone https://github.com/mse-org/mseuniverse
cd mseuniverse
/// do all your change, including new directory
git add --all
git commit --all
git push --all
```

**Reply by fredvs** [reply]
OK, done:
https://github.com/mse-org/mseuniverse/tree/main/samples/db/sample_lookup

Many thanks Med.

**Reply by medhome63** [reply]
Thanks.
So it's not possible to makedir and upload directly . Why upload in  the Menu!!

Ps : it's possible to change ncol with dropdown.valuecol and give the value in object inspector.

**Reply by fredvs** [reply]
>So it's not possible to makedir and upload directly . Why upload in the Menu!!

Are you talking about the GitHub site?
Maybe you could ask it to the GitHub site devs...

(I dont use that feature, I prefer use git push --all.)

**Reply by fredvs** [reply]
Re-hello Med.

You can create a directory with button [Add file] [Create new file]

![Screenshot_2023-09-12_21-24-56](https://github.com/mse-org/mseuniverse/assets/3421249/484042bb-0516-4681-8134-38aa51e60111)

In edit "Name your file..." just add a directory like **test**/mytest.txt

**Reply by medhome63** [reply]
New demo of sample_lookup.
[sample_lookup.zip](https://github.com/mse-org/mseuniverse/files/12689074/sample_lookup.zip)

**Comment by fredvs** [comment]
Hello Med.
Thanks for the update.
It was commited in mseuniverse: https://github.com/mse-org/mseuniverse/commit/0e04b82dd628d6d026478a4bac3d499d1606cc9a

**Reply by medhome63** [reply]
Hello Fred,
There is a new version  without tdbnavigator.datalink  propertie.
[sample_lookup.zip](https://github.com/mse-org/mseuniverse/files/12706075/sample_lookup.zip)

**Comment by fredvs** [comment]
Hello Med.

Thanks for the new demo, I will study it asap.

>There is a new version without tdbnavigator.datalink propertie.

So you found how to do without  tdbnavigator.datalink property published?
